sysl2/sysl/ints.go

```diff
@@ -0,0 +1,219 @@
+			for _, choice := range t.Alt.GetChoice() {
+				v.handleStatement(appname, epname, choice.GetStmt())
+			}
+		default:
+			panic("No statement!")
+		}
+	}
+}
+
+func (v *IntegrationParam) addAppDependency(sourceApp, sourceEndpt, targetApp, targetEndpt string, stmt *sysl.Statement) bool {
+	if len(v.apps) == 0 {
+		v.apps = v.apps.Union(v.highlights)
+	}
+	apps := MakeStrSet(sourceApp, targetApp)
+	if len(apps.Intersection(v.excludes)) > 0 {
+		return false
+	}
+	if len(apps.Intersection(v.highlights)) > 0 {
+		if !hasPattern("human", v.m.GetApps()[sourceApp].GetAttrs()) {
+			v.apps.Insert(sourceApp)
+		}
+		if !hasPattern("human", v.m.GetApps()[targetApp].GetAttrs()) {
+			v.apps.Insert(targetApp)
+		}
+	}
+	if !(apps.IsSubSet(v.apps) || (v.apps.Contains(sourceApp) && v.passthroughs.Contains(targetApp)) || v.passthroughs.Contains(sourceApp)) {
+		return false
+	}
+	if len(MakeStrSet(sourceEndpt, targetEndpt).Intersection(v.commonEndpts)) > 0 {
+		return false
+	}
+	k := fmt.Sprintf("%s:%s:%s:%s", sourceApp, sourceEndpt, targetApp, targetEndpt)
+	if _, has := v.deps[k]; has {
+		return false
+	}
+	v.deps[k] = MakeAppDependency(MakeAppElement(sourceApp, sourceEndpt), MakeAppElement(targetApp, targetEndpt), stmt)
+
+	return true
+}
+
+func (v *IntegrationParam) addEndpoint(targetApp, targetEndpt string) {
+	if targetEndpt != ".. * <- *" {
+		for epname, endpt := range v.m.GetApps()[targetApp].GetEndpoints() {
+			v.handleStatement(targetApp, epname, endpt.GetStmt())
+		}
+	}
+}
+
+func GenerateIntegrations(
+	root_model, title, output, project, filter, modules string,
+	exclude []string, clustered, epa bool,
+) map[string]string {
+	r := make(map[string]string)
+	if filter != "" {
+		re := regexp.MustCompile(filter)
+		if !re.MatchString(output) {
+			return r
+		}
+	}
+	mod := loadApp(root_model, modules)
+
+	if len(exclude) == 0 && project != "" {
+		exclude = []string{project}
+	}
+	excludeStrSet := MakeStrSet(exclude...)
+
+	// The "project" app that specifies the required view of the integration
+	app := mod.GetApps()[project]
+	of := MakeFormatParser(output)
+	// Interate over each endpoint within the selected project
+	for epname, endpt := range app.GetEndpoints() {
+		excludes := MakeStrSetFromSpecificAttr("exclude", endpt.GetAttrs())
+		passthroughs := MakeStrSetFromSpecificAttr("passthrough", endpt.GetAttrs())
+		integrations := MakeStrSetFromActionStatement(endpt.GetStmt())
+		output_dir := of.FmtOutput(project, epname, endpt.GetLongName(), endpt.GetAttrs())
+
+		highlightApps := findHighlights(mod, integrations)
+		v := MakeIntegrationParam(mod, excludeStrSet.Union(excludes), passthroughs, highlightApps)
+		v.collectApplicationDependencies(integrations)
+		intsParam := &IntsParam{v.apps.ToSlice(), v.highlights, v.deps, app, endpt}
+		args := &Args{title, project, clustered, epa}
+		r[output_dir] = GenerateView(args, intsParam, mod)
+	}
+
+	return r
+}
+
+func findHighlights(mod *sysl.Module, apps StrSet) StrSet {
```

Comment:
so is highlight any app that does not have `human` pattern?
If yes, then highlights is not a good name for this ( I know this is from legacy, but it is time to change it now). Its more like "drawable" apps.

Reply:
changed to drawableApps